tests/foreman/api/test_smartproxy.py

```diff
@@ -14,14 +14,224 @@
 
 @Upstream: No
 """
+from fauxfactory import gen_string
 from nailgun import entities
+from requests import HTTPError
+
+from robottelo.api.utils import one_to_many_names
+from robottelo.cleanup import capsule_cleanup
 from robottelo.config import settings
+from robottelo.datafactory import valid_data_list
 from robottelo.decorators import (
+    bz_bug_is_open,
+    run_only_on,
     skip_if_bug_open,
+    skip_if_not_set,
     tier1,
+    tier2,
+)
+from robottelo.helpers import (
+    default_url_on_new_port,
+    get_available_capsule_port,
 )
 from robottelo.test import APITestCase
-from robottelo.api.utils import one_to_many_names
+
+
+class CapsuleTestCase(APITestCase):
+    """Tests for Smart Proxy (Capsule) entity."""
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_url(self):
+        """Proxy creation with random URL
+
+        @id: e48a6260-97e0-4234-a69c-77bbbcde85d6
+
+        @Assert: Proxy is not created
+        """
+        # Create a random proxy
+        with self.assertRaises(HTTPError) as context:
+            entities.SmartProxy(
+                url='http://{0}:{1}'.format(
+                    gen_string('alpha', 6),
+                    gen_string('numeric', 4)
+                )).create()
+        self.assertRegexpMatches(
+            context.exception.response.text, u'Unable to communicate')
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_name(self):
+        """Proxy creation with valid name
+
+        @id: 0ffe0dc5-675e-45f4-b7e1-a14d3dd81f6e
+
+        @Assert: Proxy is created
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                with default_url_on_new_port(
+                        9090, get_available_capsule_port()) as url:
+                    proxy = entities.SmartProxy(name=name, url=url).create()
+                    self.assertEquals(proxy.read().name, name)
+                # Add capsule id to cleanup list
+                self.addCleanup(capsule_cleanup, proxy.id)
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_positive_delete(self):
```

Comment:
as for me, it doesn't make sense to add all this redundant logic. Delete functionality doesn't work so just decorate whole test case with BZ id and remove this unnecessary logic